os/hal/ports/NRF51/NRF51822/serial_lld.c

```diff
@@ -64,27 +61,67 @@ static const SerialConfig default_config = {
 /*
  * @brief Maps a baudrate speed to a BAUDRATE register value.
  */
-static uint32_t regval_from_baudrate(uint32_t speed)
+
+/**
+ * @brief   Common UART configuration.
+ *
+ */
+static void configure_uart(const SerialConfig *config)
 {
-  switch (speed) {
-  case 1200:    return 0x0004F000;
-  case 2400:    return 0x0009D000;
-  case 4800:    return 0x0013B000;
-  case 9600:    return 0x00275000;
-  case 14400:   return 0x003B0000;
-  case 19200:   return 0x004EA000;
-  case 28800:   return 0x0075F000;
-  case 38400:   return 0x009D5000;
-  case 57600:   return 0x00EBF000;
-  case 76800:   return 0x013A9000;
-  case 115200:  return 0x01D7E000;
-  case 230400:  return 0x03AFB000;
-  case 250000:  return 0x04000000;
-  case 460800:  return 0x075F7000;
-  case 921600:  return 0x0EBEDFA4;
-  case 1000000: return 0x10000000;
+  /* TODO: Add support for CTS/RTS! */
+  uint32_t speed = UART_BAUDRATE_BAUDRATE_Baud250000;
+
+  osalDbgAssert(config->rx_pin < 32, "invalid rx pin");
+  osalDbgAssert(config->tx_pin < 32, "invalid tx pin");
+
+  switch (config->speed) {
+    case 1200: speed = UART_BAUDRATE_BAUDRATE_Baud1200; break;
+    case 2400: speed = UART_BAUDRATE_BAUDRATE_Baud2400; break;
+    case 4800: speed = UART_BAUDRATE_BAUDRATE_Baud4800; break;
+    case 9600: speed = UART_BAUDRATE_BAUDRATE_Baud9600; break;
+    case 14400: speed = UART_BAUDRATE_BAUDRATE_Baud14400; break;
+    case 19200: speed = UART_BAUDRATE_BAUDRATE_Baud19200; break;
+    case 28800: speed = UART_BAUDRATE_BAUDRATE_Baud28800; break;
+    case 38400: speed = UART_BAUDRATE_BAUDRATE_Baud38400; break;
+    case 57600: speed = UART_BAUDRATE_BAUDRATE_Baud57600; break;
+    case 76800: speed = UART_BAUDRATE_BAUDRATE_Baud76800; break;
+    case 115200: speed = UART_BAUDRATE_BAUDRATE_Baud115200; break;
+    case 230400: speed = UART_BAUDRATE_BAUDRATE_Baud230400; break;
+    case 250000: speed = UART_BAUDRATE_BAUDRATE_Baud250000; break;
+    case 460800: speed = UART_BAUDRATE_BAUDRATE_Baud460800; break;
+    case 921600: speed = UART_BAUDRATE_BAUDRATE_Baud921600; break;
+    case 1000000: speed = UART_BAUDRATE_BAUDRATE_Baud1M; break;
+    default: osalDbgAssert(0, "invalid baudrate"); break;
+  };
+
+  /* Configure PINs */
+  NRF_UART0->PSELRTS = ~0;
+  NRF_UART0->PSELCTS = ~0;
+  if (config->tx_pin != NRF51_SERIAL_INVALID_PIN) {
+    // Should we assume the mode is set by the board config?

```

Comment:
If the user has to add the pins to the config anyway you can configured them here so please remove these comments.